Automattic/wp-calypso
# Checkout v2: Show discounts for each line item and introductory offer detail

The discount list in v2 checkout can sometimes create confusion because the discounts are displayed summarized far away from the line items they apply to. This is fine for simpler discounts but for complex discounts this can be impossible to understand.

## Proposed Changes

In this PR we refactor the discount list in checkout v2 to show each discount next to the line item to which it applies, with a few exceptions like coupon discounts which will still be shown below the line item list.

In addition, this PR adds additional detail to introductory offer discounts where the billing term length of the discount is different from the term length of the product. For example, a 3 month discount for an annual product. This discount is misleading because while the user may indeed be "saving" $X today, they will owe ($X / 12) * 3 in 3 months. This PR hides those discount amounts, instead showing the line item price crossed-out with today's price adjacent to it (or just the crossed-out price if the discount is 100%). It also adds text like `Due XXXX-MM-YY: $X` below the line item to clarify when and how much the user will next be charged.

Before             |  After
:-------------------------:|:-------------------------:
<img width="259" alt="before-another" src="https://github.com/Automattic/wp-calypso/assets/2036909/801e474a-df26-41f7-ab43-c618a59750b5"> | <img width="257" alt="after-another" src="https://github.com/Automattic/wp-calypso/assets/2036909/134f1e37-983a-4630-9d8a-ad73bd74d68b">

Fixes https://github.com/Automattic/payments-shilling/issues/2410
Fixes https://github.com/Automattic/payments-shilling/issues/2437

## Testing Instructions

- Add a plan, a domain, and a Professional email product to your cart and visit checkout.
- Add a coupon to the cart.
- Compare checkout (v1) on this branch to production and verify that there are no changes.
- Add `?checkoutVersion=2` to the checkout URL on this branch and verify that the changes described above look correct, particularly the "Due" date and price.
